lib/msal-browser/src/interaction_client/BaseInteractionClient.ts

```diff
@@ -91,6 +91,11 @@ export abstract class BaseInteractionClient {
             this.logger.verbose(`Authentication Scheme set to "${request.authenticationScheme}" as configured in Auth request`);
         }
 
+        // Set requested claims hash if claims were requested
+        if (request.claims && !StringUtils.isEmpty(request.claims)) {
+            request.requestedClaimsHash = await this.browserCrypto.hashString(request.claims);
+        } 
+
         const validatedRequest: BaseAuthRequest = {
```

Comment:
Instead of having this method return `BaseAuthRequest`, can we have it return some internal type that includes `requestedClaimsHash`?

Reply:
That would be one way to pass this downstream without exposing it upstream.

Reply:
The other alternative is to have it omitted from `SilentRequest`, `RedirectRequest`, `PopupRequest`, etc, which there is already precedence for, so I'd be fine with either approach.

Reply:
Got it, I'll go through both options and see which is cleaner.